### PR TITLE
[FIX] quality_control: Fix TypeError on new test category

### PR DESCRIPTION
**Pull request by zamberjo**
`TestQualityControl` is failing on `QcTestTemplateCatergory._get_complete_name`
when `name` is set as `False`.

### CODE REVIEW

**Comment by oihane** [comment]
If `name` is required how can it be `False`?

**Comment by zamberjo** [comment]
Is failling on line:20 (`self.complete_name = " / ".join(reversed(names))`) when computing values for new records (not yet persisted) as `join` is expeacting a list of strings and receives a list of booleans (`[False]`)

**Comment by oihane** [comment]
Won't be more correct not to compute the `complete_name` if there is not `name`?

**Comment by zamberjo** [comment]
LGTM

**Review by oihane — APPROVED** [review]
LGTM

**Review by agaldona — APPROVED** [review]

File: quality_control/models/qc_test_category.py [hunk 1]
```diff
@@ -12,12 +12,13 @@ class QcTestTemplateCategory(models.Model):
     @api.one
     @api.depends('name', 'parent_id')
     def _get_complete_name(self):
-        names = [self.name]
-        parent = self.parent_id
-        while parent:
-            names.append(parent.name)
-            parent = parent.parent_id
-        self.complete_name = " / ".join(reversed(names))
+        if self.name:
```
Comment (pedrobaeza) [inline quality_control/models/qc_test_category.py]: Please don't make the diff bigger and increase cyclomatic complexity. Make instead:
```
if not self.name:
    return
...
```

Comment (oihane) [reply quality_control/models/qc_test_category.py]: Wouldn't be more correct to convert it to `@api.multi`
```python
    for record in self.filtered('name'):
        names = [record.name]
        parent = record.parent_id
        while parent:
            names.append(parent.name)
            parent = parent.parent_id
        record.complete_name = " / ".join(reversed(names))     
```

**Comment by LoisRForgeFlow** [comment]
Just a favor, can you squash to only one commit when merging?  :grin:, that way is very fast for me to move this changes to v9 PR.

**Comment by MiquelRForgeFlow** [comment]
Please @zamberjo , will you update the PR?

**Review by damendieta — APPROVED** [review]

File: quality_control/models/qc_test_category.py (hunk 1 above)
Comment (damendieta) [reply quality_control/models/qc_test_category.py]: I think it's better to just add a or '' to the code, like this:

names = [record.name or '']

But it works, so, approved.

**Comment by pedrobaeza** [comment]
I think one of the suggestions should be applied at least before merging

**Comment by damendieta** [comment]
@zamberjo are you going to apply the suggestions.

As @zamberjo hasn't commented or made any commit from may 10, maybe it's better to close this PR and open a new one with a fix.

It's a one liner 3 characters fix and it's hanging for almost 6 months.

Please @zamberjo, could you be so kind to help us with this modifications?

Thanks.

**Comment by LoisRForgeFlow** [comment]
I agree with @damendieta , this is a very small fix and it has to move forward... 

EDIT: Please, @damendieta open a PR with your proposal and we can supersede this PR.

**Comment by zamberjo** [comment]
Sorry, I plan to review all outstanding PRs -including this one- over the week.

**Comment by LoisRForgeFlow** [comment]
@zamberjo ok, let's keep this one then.

**Comment by damendieta** [comment]
@zamberjo thanks a lot.
If possible, it's better to fix it just with this:

names = [record.name or '']

Adding the empty string alternative will solve the problem with the least code and no need to add a new indent to the formula, also, only one line will change.

Could you please consider this fix instead.

Thanks a lot.

**Comment by zamberjo** [comment]
I should change the method name `_get_complete_name` to `_compute_complete_name`?

**Review by damendieta — APPROVED** [review]

**Pull request merged**